contract-message-predicate/src/contract_message_predicate.sw

```diff
@@ -0,0 +1,82 @@
+predicate;
+
+dep transaction_utils;
+
+use std::{assert::assert, constants::ZERO_B256};
+use transaction_utils::{
+    input_coin_amount,
+    input_coin_asset_id,
+    input_contract_contract_id,
+    input_count,
+    input_message_data,
+    input_message_data_length,
+    output_contract_input_index,
+    output_count,
+    tx_gas_limit,
+    tx_gas_price,
+    tx_script_bytecode_hash,
+    verify_input_coin,
+    verify_input_contract,
+    verify_input_message,
+    verify_output_change,
+    verify_output_contract,
+    verify_output_variable,
+};
+
+///////////////
+// CONSTANTS //
+///////////////
+// The minimum gas limit for the transaction not to revert out-of-gas
+// TODO: research what gas amount is reasonable and possibly move to config time
+const MIN_GAS = 1_200_000;
+
+// The hash of the script which must spend the input belonging to this predicate
+const SPENDING_SCRIPT_HASH = 0x054db8e175fcbbb08c5702d0ba274918d1f1bfac703580ffd9cdac1074e0d808;
+
+// The input and output index values
+const INPUT_CONTRACT_INDEX = 0u8;
```

Comment:
Should these be in the `std`?

Reply:
No, they are specific to this particular predicate and script setup